trip_reservations/src/main/kotlin/pl/edu/pg/rsww/tripreservations/TripReservationsController.kt

```diff
@@ -1,40 +1,75 @@
 package pl.edu.pg.rsww.tripreservations
 
-import kotlinx.serialization.decodeFromString
+import com.mongodb.client.model.Filters.and
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.client.model.Filters.lt
+import com.mongodb.client.model.Updates.inc
+import com.mongodb.kotlin.client.MongoClient
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
+import org.bson.Document
 import org.springframework.amqp.core.Message
-import org.springframework.amqp.rabbit.annotation.Queue
-import org.springframework.amqp.rabbit.annotation.RabbitListener
 import org.springframework.amqp.rabbit.core.RabbitTemplate
 import org.springframework.beans.factory.annotation.Autowired
 
+data class Entity(
+    val entity_id: String,
+    val entity_type: String,
+    val last_event_id: Int,
+    val data: Document,
+)
+
 public class TripReservationController(
     @Autowired private val template: RabbitTemplate,
     @Autowired private val queueConfig: QueueConfig,
 ) {
     private val activeOrchestrators = HashMap<String, TripReservationOrchestrator>()
 
+    fun updateReservationCounter(
```

Comment:
We also need to create an event in the SQL database to do event sourcing. That should happen before creating (not updating) a snapshot with new `last_event_id` in the MongoDB.